### PR TITLE
Refactor of main to allow graceful shutdown

### PR DESCRIPTION
**Pull request by r4f4ss**
solves #156

### CODE REVIEW

**Comment by GrapeBaBa** [comment]
Since we forked geth, is there any existing code to reuse or reference.

File: cmd/shisui/main.go [hunk 1]
```diff
@@ -45,6 +47,14 @@ type Config struct {
 	Networks     []string
 }
 
+type Client struct {
+	DiscV5API      *discover.DiscV5API
```
Comment (fearlessfe) [inline cmd/shisui/main.go:51]: `DiscV5API` is not needed to close. In fact, all network use the same `UDPv5`, every network will stop the discv5 when call the Stop method

**Comment by r4f4ss** [comment]
> Since we forked geth, is there any existing code to reuse or reference.

The reference comes from https://github.com/optimism-java/shisui/blob/b061173db18614af29b49b7dbab4186cd5d92497/cmd/utils/cmd.go#L77 that geth main function calls to create a new node. Inside this function (```StartNode```) a goroutine watches a channel for signals of interruption and executes the shutdown function (https://github.com/optimism-java/shisui/blob/b061173db18614af29b49b7dbab4186cd5d92497/cmd/utils/cmd.go#L115).

I studied this file to understand how to handle a ```syscall.SIGINT``` or a ```syscall.SIGTERM``` signals, but I believe it is not possible to reuse this code to deal with a Portal Network since geth implements an anonymous goroutine.

Other aspect is the behavior/functionality of CTRL-C command: in geth to force quit is necessary to press 10 times CTRL-C, or panic behavior:
```go
for i := 10; i > 0; i-- {
        <-sigc
        if i > 1 {
          log.Warn("Already shutting down, interrupt more to panic.", "times", i-1)
        }
      }
```

while in Shisui only a second press of CTRL-c quits.

Please let me know if there is a better approach I should implement.

File: cmd/shisui/main.go (hunk 1 above)
Comment (r4f4ss) [reply cmd/shisui/main.go:51]: I tried to move the closure of ```DiskV5``` from ```PortalProtocol```, where it was originally, to ```main```, but it broke a test and may potentially break other parts of the code that calls ```PortalProtocol.Close()```.

```DiscV5``` close uses a ```sync.Once``` struct and can safely be closed many times: https://github.com/optimism-java/shisui/blob/b061173db18614af29b49b7dbab4186cd5d92497/p2p/discover/v5_udp.go#L193

**Review by GrapeBaBa — APPROVED** [review]

**Pull request merged**